### PR TITLE
Better handling of task variables in SchedulingTaskComparator

### PR DESCRIPTION
**Pull request by fviale**
- replace contents of variables inside scripts : when two tasks have the same selection script and use the same variable. The two tasks are compatible when the variable has the same value.
 - minor : add equals to ParallelEnvironment
 - minor : in SchedulingMethodImpl, moved the code to initialize eligible tasks

### CODE REVIEW

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

1. MINOR [SchedulingMethodImpl.java#L499](https://github.com/ow2-proactive/scheduling/blob/4053f8fd9bf1431bf1c6b421914ab6070212a688/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java#L499): Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.lang.ClassNotFoundException [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1160)
1. MINOR [SchedulingTaskComparator.java#L53](https://github.com/ow2-proactive/scheduling/blob/4053f8fd9bf1431bf1c6b421914ab6070212a688/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingTaskComparator.java#L53): Make the "logger" logger private static final. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1312)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

1. MINOR [SchedulingMethodImpl.java#L499](https://github.com/ow2-proactive/scheduling/blob/a04220a5f58ae05de47fd7cc1dda51c29591a70a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java#L499): Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.lang.ClassNotFoundException [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1160)
1. MINOR [SchedulingTaskComparator.java#L54](https://github.com/ow2-proactive/scheduling/blob/a04220a5f58ae05de47fd7cc1dda51c29591a70a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingTaskComparator.java#L54): Make the "logger" logger private static final. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1312)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingTaskComparator.java [hunk 1]
```diff
@@ -53,7 +62,28 @@
...
+
+            for (SelectionScript script : scriptList) {
+                SelectionScript modifiedScript = script;
+                try {
+                    Map<String, Serializable> bindings = SchedulingMethodImpl.createBindingsForSelectionScripts(job,
+                                                                                                                task);
+                    modifiedScript = SchedulingMethodImpl.replaceBindingsInsideScript(script, bindings);
+
+                } catch (Exception e) {
+                    logger.error("", e);
```
Comment (marcocast) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingTaskComparator.java]: can we add more info in this error log?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingTaskComparator.java [hunk 2]
```diff
@@ -53,7 +62,28 @@
...
+                }
+                try {
+                    digests.add(new String(modifiedScript.digest()));
+                } catch (Exception e) {
+                    logger.error("", e);
```
Comment (marcocast) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingTaskComparator.java]: same here for this error log

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingTaskComparator.java (hunk 1 above)
Comment (paraita) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingTaskComparator.java]: like the name of the task/job so it'd be easier to locate what selection script failed from the studio.

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue

1. MINOR [SchedulingMethodImpl.java#L499](https://github.com/ow2-proactive/scheduling/blob/53841d15d8faaee28f240d2b1a28385f475cd3b3/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java#L499): Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.lang.ClassNotFoundException [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1160)

**Pull request merged**